tool/client/transformers/dispense-claim.ts

```diff
@@ -0,0 +1,284 @@
+
+const CODEABLE_CONCEPT_PRESCRIPTION_CHARGE_NOT_PAID: fhir.CodeableConcept = {
+  coding: [
+    {
+      system: "https://fhir.nhs.uk/CodeSystem/DM-prescription-charge",
+      code: "not-paid",
+      display: "Not Paid"
+    }
+  ]
+}
+
+export function createClaim(prescriptionOrder: fhir.Bundle, dispenseNotifications: Array<fhir.Bundle>): fhir.Claim {
+  const medicationRequests = getMedicationRequestResources(prescriptionOrder)
+  const medicationDispenses = dispenseNotifications.map(getMedicationDispenseResources)
+    .reduce((a, b) => a.concat(b), [])
+  const patient = getPatientResources(prescriptionOrder)
+
+  const patientIdentifier = patient[0].identifier[0]
+
+  const finalMedicationDispense = medicationDispenses[medicationDispenses.length - 1]
+  const prescriptionStatusExtension = finalMedicationDispense.extension.find(
+    e => e.url === "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-TaskBusinessStatus"
+  ) as fhir.CodingExtension
+
+  const claimingOrganizationReference = finalMedicationDispense.performer
+    .map(performer => performer.actor)
+    .find(actor => actor.type === "Organization")
+
+  const groupIdentifierExtension = finalMedicationDispense.authorizingPrescription[0].extension.find(
+    e => e.url === "https://fhir.nhs.uk/StructureDefinition/Extension-DM-GroupIdentifier"
+  ) as fhir.GroupIdentifierExtension
+
+  return {
+    resourceType: "Claim",
+    created: new Date().toISOString(),
+    identifier: [createClaimIdentifier()],
+    status: "active",
+    type: CODEABLE_CONCEPT_CLAIM_TYPE_PHARMACY,
+    use: "claim",
+    patient: createClaimPatient(patientIdentifier),
+    provider: createClaimProvider(),
+    priority: CODEABLE_CONCEPT_PRIORITY_NORMAL,
+    insurance: [INSURANCE_NHS_BSA],
+    payee: createClaimPayee(claimingOrganizationReference),
+    prescription: createClaimPrescription(groupIdentifierExtension),
+    item: [createClaimItem(prescriptionStatusExtension, medicationRequests, medicationDispenses)]
+  }
+}
+
+function createClaimIdentifier(): fhir.Identifier {
+  return {
+    system: "https://tools.ietf.org/html/rfc4122",
+    value: uuid.v4()
+  }
+}
+
+function createClaimPatient(identifier: Identifier) {
+  return {
+    //Doing it this way to avoid copying the verification status extension
+    identifier: {
+      system: identifier.system,
+      value: identifier.value
+    }
+  }
+}
+
+function createClaimProvider() {
+  //TODO - is this actually needed?
+  return {
+    identifier: {
+      system: "https://fhir.nhs.uk/Id/sds-role-profile-id",
+      value: "999999999999"
+    },
+    display: "Unattended Access"
+  }
+}
+
+function createClaimPayee(claimingOrganizationReference: fhir.IdentifierReference<fhir.Organization>): fhir.ClaimPayee {
+  return {
+    type: CODEABLE_CONCEPT_PAYEE_TYPE_PROVIDER,
+    party: claimingOrganizationReference
+  }
+}
+
+function createClaimPrescription(groupIdentifierExtension: fhir.GroupIdentifierExtension): fhir.ClaimPrescription {
```

Comment:
not ur code but i hate how this is in an extension. would much prefer to use the `https://fhir.nhs.uk/Id/prescription-order-item-number` identifier

Reply:
Yeah, needs to be a reference to the overall prescription, not just a line item